splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/TernaryFunctionForBitDataIT.java

```diff
@@ -0,0 +1,375 @@
+                "-------------------------\n" +
+                " 3 |-abac bc- |-bc- | 6 |";
+
+        for (String col1: col1set) {
+            for (String col2 : col2set) {
+                try {
+                    /* Q1 */
+                    String sqlText = format("select id, '-' || a2 || '-', '-' || b2 || '-', locate(%1$s,%2$s) from t1 --splice-properties useSpark=%3$s\n order by id", col2, col1, useSpark);
+
+                    ResultSet rs = methodWatcher.executeQuery(sqlText);
+                    Assert.assertEquals("\nlocate(" + col2 + "," + col1 + "):" + sqlText + "\n", expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+                    rs.close();
+
+                    /* Q2 */
+                    sqlText = format("select id, '-' || a2 || '-', '-' || b2 || '-', locate(%1$s,%2$s) from t1 --splice-properties useSpark=%3$s\n where locate(%1$s,%2$s)>1 order by id", col2, col1, useSpark);
+
+                    rs = methodWatcher.executeQuery(sqlText);
+                    Assert.assertEquals("\nlocate(" + col2 + "," + col1 + "):" + sqlText + "\n", expected2, TestUtils.FormattedResult.ResultFactory.toString(rs));
+                    rs.close();
+                } catch (SQLSyntaxErrorException e) {
+                    // it is expected that long varchar and clob are not compared with bit data type
+                    if ((col1.equals("a3") || col1.equals("a4")) && (col2.equals("b6") || col2.equals("b7") || col2.equals("b8")) ||
+                        (col2.equals("b3") || col2.equals("b4")) && (col1.equals("a6") || col1.equals("a7") || col1.equals("a8")))
+                        Assert.assertEquals("locate(" + col2 + "," + col1 + "):" + e.getMessage(), SQLState.LANG_DB2_FUNCTION_INCOMPATIBLE, e.getSQLState());
+                    else
+                        Assert.fail("locate(" + col2 + "," + col1 + "):" + e.getMessage());
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testLocateParameterizedQuery1() throws Exception {
+        ResultSet rs = null;
+        String sqlText = format("select id, '-' || a2 || '-', locate(?, a5) from t1 --splice-properties useSpark=%s\n order by id", useSpark);
+        try {
+            PreparedStatement ps = methodWatcher.prepareStatement(sqlText);
+            ps.setBytes(1, "ac".getBytes());
+            rs = ps.executeQuery();
+
+            String expected = "ID |    2     |  3  |\n" +
+                    "---------------------\n" +
+                    " 1 |-abac bc- |  3  |\n" +
+                    " 2 |-abac bc- |  3  |\n" +
+                    " 3 |-abac bc- |  3  |\n" +
+                    " 4 |  -ab-    |  0  |\n" +
+                    " 5 |   --     |  0  |\n" +
+                    " 6 |  -ab-    |  0  |\n" +
+                    " 7 |  NULL    |NULL |";
+
+            Assert.assertEquals("\n" + sqlText + "\n", expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        } finally {
```

Comment:
You can use a try-with-resources to simplify this:
try (ResultSet rs = ps.executeQuery()) {
}
It will automatically close the resultset for you.

Reply:
Done!